### PR TITLE
test: make load and resolve-extends resolve ESM modules correctly

### PR DESCRIPTION
**Pull request by marcalexiei**
## Description

`resolve-extends` now provides function to resolve configuration and parser that are used by himself and `load`

---

Additional changes

- `vitest` utilities are now explicitly imported
- `@jest/globals` dependencies removed
- coverage limited to `@commitlint` packages
- fix a syntax error that was breaking `@commitlint/config-nx-scopes`

## Motivation and Context

https://github.com/conventional-changelog/commitlint/pull/3850

## Usage examples

N/A

## How Has This Been Tested?

```sh
yarn test
```

### CODE REVIEW

**Comment by marcalexiei** [comment]
A quick note: consider to rebase and merge the PR instead of squash commits. 
So if there are issue they can be reverted easily 😅

File: @alias/commitlint/cli.test.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import {test, expect} from 'vitest';
```
Comment (JounQin) [inline @alias/commitlint/cli.test.js:1]: I don't know whether it is the preferred way for @escapedcat, but I'm totally fine with this change personally.

File: package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
     "publish": "lerna publish --conventional-commits",
     "reinstall": "yarn clean && yarn install",
     "start": "yarn watch",
-    "test": "yarn build && cross-env HOME=$PWD LANG=en_US.UTF-8 NO_COLOR=1 vitest run --coverage",
+    "test": "yarn build && cross-env HOME=$PWD LANG=en_US.UTF-8 vitest run --coverage",
```
Comment (JounQin) [inline package.json]: @marcalexiei

removing `NO_COLOR=1` is a bit unfriendly to me personally, I always have different snapshot if the env disabled.

cc @escapedcat

File: @alias/commitlint/cli.test.js (hunk 1 above)
Comment (escapedcat) [reply @alias/commitlint/cli.test.js:1]: Switchting to vitest? Sure, go for it! ❤️

Comment (JounQin) [reply @alias/commitlint/cli.test.js:1]: Nope, I mean global `test`, `expect` or import them manually.

I was using https://github.com/unplugin/unplugin-auto-import previously.

**Comment by JounQin** [comment]
> A quick note: consider to rebase and merge the PR instead of squash commits. So if there are issue they can be reverted easily 😅

<del>No worry, I think squashing here and rebasing from upstream is the correct/easiest way to go.</del>

---

OK, I decided to rebase instead, let's see what will happen.

File: @alias/commitlint/cli.test.js (hunk 1 above)
Comment (escapedcat) [reply @alias/commitlint/cli.test.js:1]: Ah, I don't mind using auto-import

File: package.json (hunk 2 above)
Comment (marcalexiei) [reply package.json]: I removed `NO_COLOR=1` to have debug log more readable. I can restore it.

File: @alias/commitlint/cli.test.js (hunk 1 above)
Comment (JounQin) [reply @alias/commitlint/cli.test.js:1]: Let's keep this PR as-is, and figure  out the best way to use vitest later.

**Pull request merged**